lib/offsite_payments/integrations/latipay.rb

```diff
@@ -0,0 +1,247 @@
+module OffsitePayments
+  module Integrations
+    module Latipay
+
+      def self.notification(post, options = {})
+        Notification.new(post, options)
+      end
+
+      def self.return(query_string, options = {})
+        Return.new(query_string, options)
+      end
+
+      class Interface
+        include ActiveUtils::PostsData # ssl_get/post
+
+        def self.base_url
+          "https://api.latipay.net/v2"
+        end
+
+        def initialize(api_key, user_id)
+          @api_key = api_key
+          @user_id = user_id
+        end
+
+        def sign(fields)
+          message = fields.compact.sort.map{ |k,v| "#{k.to_s}=#{v}" }.join('&').concat(@api_key)
```

Comment:
This looks like we're manually creating some sort of query params? Should we do some real escaping and lean on Ruby to do this for us?

https://ruby-doc.org/stdlib-2.7.1/libdoc/uri/rdoc/URI.html#method-c-encode_www_form

Reply:
I'm not sure why, I would guess Jing didn't think of that since he's generating a signing field here instead of form parameters (thats built by whatever calls this)

Reply:
I'm inclined to leave it this way since you can see a direct correlation to the steps required in the latipay API
http://doc.latipay.net/v2/merchant-hosted-offline.html#1-Latipay-Transaction-Interface SHA-256 HMAC Signature